biz.aQute.resolve/src/biz/aQute/resolve/BndrunResolveContext.java

```diff
@@ -371,6 +380,61 @@ private void loadRepositories() throws Exception {
 		}
 	}
 
+	private void findAdditionalAugments(Parameters augments, Collection<Repository> orderedRepositories) {
+		RequirementBuilder rb = new RequirementBuilder("bnd.augment");
+		rb.filter("(path=*)");
+		Requirement req = rb.buildSyntheticRequirement();
+
+		for (Repository r : orderedRepositories) {
+			Map<Requirement,Collection<Capability>> found = r.findProviders(Collections.singleton(req));
+			Collection<Capability> capabilities = found.get(req);
+			if (capabilities != null) {
+				for (Capability capability : capabilities) {
+					findAdditionalAugmentsFromResource(augments, capability);
+				}
+			}
+		}
+	}
+
+	private void findAdditionalAugmentsFromResource(Parameters augments, Capability capability) {
+		Resource resource = capability.getResource();
+		Map<URI,String> locations = ResourceUtils.getLocations(resource);
+
+		if (locations == null || locations.isEmpty())
+			return;
+
+		Object pathObject = capability.getAttributes().get("path");
+		if (pathObject == null)
+			pathObject = "augments.bnd";
+
+		if (pathObject instanceof String) {
+			String path = (String) pathObject;
+
+			HttpClient http = project.getWorkspace().getPlugin(HttpClient.class);
```

Comment:
This should be registry.getPlugin, no? All other plugins come from the registry field.